tests/components/utils/renderToContainer.test.tsx

```diff
@@ -0,0 +1,23 @@
+import renderToContainer from '../../../src/components/utils/renderToContainer';
+import packageJson from '../../../package.json';
+
+jest.mock('react-dom/client');
+jest.mock('react-dom');
+
+/**
+ * This test needs to pass for all react-dom versions we support.
+ */
+it('can render a react component into native dom elements', async () => {
+  const createRootRender = jest.fn();
+  const createRootSpy = jest.spyOn(require('react-dom/client'), 'createRoot')
+    .mockImplementation(() => ({ render: createRootRender }));
+  const reactDomRenderSpy = jest.spyOn(require('react-dom'), 'render');
+
+  const container = document.createElement('div');
+
+  await renderToContainer(<span>test me</span>, container);
+  const isReactDom18 = packageJson.devDependencies['react-dom'].match(/18\.\d*\.\d*/);
```

Comment:
do you have to change the react version on your machine to test each case?

Reply:
our ci takes care of that since it runs in all react 16 17 and 18
https://github.com/yext/search-ui-react/blob/main/.github/workflows/run-tests.yml
but if you wanted to test it locally then yeah

Reply:
ci covering that sounds good to me